src/Atc.Rest.Client/Builder/MessageRequestBuilder.cs

```diff
@@ -56,56 +56,48 @@ public IMessageRequestBuilder WithBody<TBody>(TBody body)
             return this;
         }
 
-        public IMessageRequestBuilder WithPathParameter(string name, string value)
+        public IMessageRequestBuilder WithPathParameter(string name, object? value)
         {
             if (string.IsNullOrWhiteSpace(name))
             {
                 throw new ArgumentException($"'{nameof(name)}' cannot be null or whitespace", nameof(name));
```

Comment:
```suggestion
                throw new ArgumentNullException($"'{nameof(name)}' cannot be null or whitespace", nameof(name));
```

Pr. the interface docs, this should be the correct exception to throw here.